erpnext/accounts/doctype/sales_invoice/sales_invoice.js

```diff
@@ -767,8 +767,7 @@ frappe.ui.form.on('Sales Invoice', {
 
 	hide_fields: function(frm) {
 		let doc = frm.doc;
-		var parent_fields = ['project', 'due_date', 'is_opening', 'source', 'total_advance', 'get_advances',
```

Comment:
Use `depends_on` to hide the advance related fields if `set_advances` is not checked or `advances` doesn't have rows